File: source/back/BankInformationSystem.Data/Entities/Contacts.cs
namespace BankInformationSystem.Data.Entities
{
    public class Contacts
    {
        public int Id { get; set; }
        
        public string Email { get; set; }

        public string HomePhoneNumber { get; set; }

        public string MobilePhoneNumber { get; set; }
        
        public int CustomerId { get; set; }

        public Customer Customer { get; set; }
    }
}

File: source/front/src/models/entities/Contacts.ts
import { Stateful, action } from "reactronic"

export interface IContacts {
  email: string
  homePhoneNumber: string
  mobilePhoneNumber: string
}
export class Contacts extends Stateful {
  email: string = ''
  homePhoneNumber: string = ''
  mobilePhoneNumber: string = ''

  @action
  initialize(contacts: IContacts): void {
    this.email = contacts.email
    this.homePhoneNumber = contacts.homePhoneNumber
    this.mobilePhoneNumber = contacts.mobilePhoneNumber
  }

  @action
  setEmail(value: string): void {
    this.email = value
  }

  @action
  setHomePhoneNumber(value: string): void {
    this.homePhoneNumber = value
  }

  @action
  setMobilePhoneNumber(value: string): void {
    this.mobilePhoneNumber = value
  }
}
